packages/react-search-ui/src/types/ResultWithRequiredId.js

```diff
@@ -0,0 +1,18 @@
+import PropTypes from "prop-types";
+
+/*
+  [TODO Next Major]
+
+  This file exists alongside Result.js to enforce that Result objects have an
+  ID in certain views that already require it.
+
+  I wanted to add id directly to Result.js, but that would be a breaking change. This exists
+  to give better validation warnings to those files that already implicitly required it.
+*/
```

Comment:
So just to check my understanding:

1. Are we going back to `Result` (with a required ID) the next major/breaking release?

2. What (if anything) is still using `Result` (vs `ResultWithRequiredId`)? (This is just me being curious)